paperqa/agents/search.py

```diff
@@ -496,16 +511,24 @@ async def process_file(
                     fields=["title", "author", "journal", "year"],
                     settings=settings,
                 )
-            except (ValueError, ImpossibleParsingError):
+            except Exception as e:
+                # We handle any exception here because we want to save_index so we
+                # 1. can resume the build without rebuilding this file if a separate
+                # process_file invocation leads to a segfault or crash.
+                # 2. don't have deadlock issues after.
                 logger.exception(
                     f"Error parsing {file_location}, skipping index for this file."
                 )
                 await search_index.mark_failed_document(file_location)
-                # Save so we can resume the build without rebuilding this file if a
-                # separate process_file invocation leads to a segfault or crash
                 await search_index.save_index()
                 if progress_bar_update:
                     progress_bar_update()
+
+                # TODOC: We silence ValueError because ...
+                # We silence ImpossibleParsingError because we won't be able to
+                # parse the file if it failed one time
+                if not isinstance(e, ValueError | ImpossibleParsingError):
```

Comment:
```suggestion
                if not isinstance(e, ValueError | ImpossibleParsingError):
                    # ImpossibleParsingError: parsing failure, don't retry
                    # ValueError: TODOC
```

[style nit] comments go to the right of or inside `if`, not outside